docs/support_files/world.md

```diff
@@ -36,32 +51,85 @@ $ cucumber-js --world-parameters '{"appUrl":"http://localhost:3000/"}'
+
+Let's walk through a typical scenario, setting up world that manages a browser context. We'll use the ES6 module syntax for this example.  First, let's set up our custom world. Class files should not be loaded as steps - they should be imported. So in this example we'll presume it is in a classes folder next to the steps folder.
+
+###### CustomWorld.js
+```javascript
+import { World } from '@cucumber/cucumber';
+import seleniumWebdriver from "selenium-webdriver";
+
+export default class extends World {
+  driver = null;
+
+  /*
+   * A constructor is only needed if you have custom actions
+   * to take after the Cucumber parses the options.
+   */
+  constructor(options) {
+    super(options);
+    // Custom actions go here.
+  }
+
+  /*
+   * Constructors cannot be asynchronous! To work around this we'll
+   * use an init method with the Before hook
+   */
+  async init(scenario) {
+    this.driver = await seleniumWebdriver.Builder()
+      .forBrowser(this.params.browser)
+      .build();
+  }
+}
+```
+
+Now we'll use a step file to setup this custom world and declare the before hook.
 
-You might also want to have methods on your World that hooks and steps can access to keep their own code simple. To do this, you can provide your own World class with its own properties and methods that help with your instrumentation, and then call `setWorldConstructor` to tell Cucumber about it. You should extend the built-in `World` class:
+###### setup.js
+```javascript
+import { Before, setWorldConstructor } from '@cucumber/cucumber';
+import CustomWorld from "../classes/CustomWorld.js"
+
+setWorldConstructor(CustomWorld);
+
+Before(async function(scenario) {
+  this.init(scenario);
+});
+```
 
+Custom world classes can also be used as a point to anchor methods used by multiple steps by using the prototype of the custom world. For example, the login step of your application should contain the login process:
+
+###### login.js
 ```javascript
-const { setWorldConstructor, World } = require('@cucumber/cucumber')
-const seleniumWebdriver = require('selenium-webdriver')
-
-class CustomWorld extends World {
-    driver = new seleniumWebdriver.Builder()
-        .forBrowser('firefox')
-        .build()
-
-    constructor(options) {
-        // needed so `attach`, `log` and `parameters` are properly set
-        super(options)
-    }
-
-    // Returns a promise that resolves to the element
-    async waitForElement(locator) {
-        const condition = seleniumWebdriver.until.elementLocated(locator)
-        return await this.driver.wait(condition)
-    }
+import { Given } from '@cucumber/cucumber';
+import CustomWorld from "../classes/CustomWorld.js"
+
+Given("I am an {string}", function(userType) {
+  this.login(userType)
+})
+
+CustomWorld.prototype.login(userType) {
```

Comment:
This is one possible pattern of several, and it's beyond the scope of this document to be recommending any particular one - would be an interesting blog post though!

Here, I think it's enough to demonstrate methods being called from steps/hooks on a custom world, and noting that as your world does more you'll need to think about how to manage that.

Reply:
Agreed and removed.